geopandas/io/arrow.py

```diff
@@ -853,11 +872,12 @@ def _convert_bbox_to_parquet_filter(bbox, bbox_column_name):
 
 
 def _check_if_covering_in_geo_metadata(geo_metadata):
-    return "covering" in geo_metadata["columns"]["geometry"].keys()
+    return "covering" in next(iter(geo_metadata["columns"].values()))
```

Comment:
There is also no guarantee it is the first, so you can do 

```suggestion
    primary_column = geo_metadata["primary_column"]
    return "covering" in geo_metadata["columns"][primary_column].keys()
```

That's the approach I am taking in https://github.com/geopandas/geopandas/pull/3314